XenAdmin/Wizards/PatchingWizard/PatchingWizard_PrecheckPage.cs

```diff
@@ -949,7 +955,28 @@ private void UpdateActionProgress(AsyncAction action)
 
         private void checkBox1_CheckedChanged(object sender, EventArgs e)
         {
-            RefreshRechecks();
+            RefreshDataGridViewList();
+        }
+
+        private void RefreshDataGridViewList()
+        {
+            dataGridView1.Rows.Clear();
+
+            foreach (var row in allRows)
+            {
+                if(row is PreCheckHostRow)
+                {
+                    var r = row as PreCheckHostRow;
+                    if(r.Problem != null || (r.Problem == null && !checkBoxViewPrecheckFailuresOnly.Checked))

```

Comment:
`r.Problem == null` in not needed, as it will be always true. So this can be simplified to 
```if (r.Problem != null || !checkBoxViewPrecheckFailuresOnly.Checked)```